sample_apps/run_example.sh

```diff
@@ -647,11 +1149,64 @@ function run_steps() {
 # Do initial setup of the sample app test-case
 # ###########################################################################
 function setup() {
-    run_steps "show_env" "provision_app_service_files"
+    local app_name="$1"
+    case "${app_name}" in
+        "simple_app")
+            setup_simple_app
+            ;;
+
+        "simple_app_under_oe")
+            setup_simple_app_for_oe ""
+            ;;
+
+        *)
+            echo "${Me}: Unknown sample app name '${app_name}'."
+            return
+            ;;
+    esac
+}
+
+# ###########################################################################
+# Do initial setup of the sample app test-case
+# As the sub-steps array has nested fn-calls, run a collection of steps
+# to avoid re-running sub-fns nested under author_policy()
+# ###########################################################################
+function setup_simple_app() {
+    run_steps "show_env" "get_measurement_of_trusted_app"
+    author_policy
```

Comment:
Because `author_policy` already calls some sub-step fns, we do break-up the execution to separately execute `author_policy`. Then, continue -after- the list of sub-step fns it includes.

Same gymnastics are done in `setup_simple_app_for_oe()` on L1194 onwards.